File: 1993-sum-of-all-subset-xor-totals/sum-of-all-subset-xor-totals.cpp
class Solution {
public:

    int solve(int ind,vector<int>&nums,int res){
            if(ind==nums.size()){
                return res;
            }

            int pick =solve(ind+1,nums,res^nums[ind]);
            int notpick =solve(ind+1,nums,res);

            return pick+notpick;

    }



    int subsetXORSum(vector<int>& nums) {
        return solve(0,nums,0);
    }
};

File: 1993-sum-of-all-subset-xor-totals/sum-of-all-subset-xor-totals.py
class Solution:
    def subsetXORSum(self, nums: List[int]) -> int:

        n=len(nums)

        def solve (ind:int,nums:List[int],res:int)->int:
            
            if ind==len(nums):
                return res
        
            pick=solve(ind+1,nums,nums[ind]^res)
            npick=solve(ind+1,nums,res)

            return pick+npick
        
        return solve(0,nums,0)